### DISCUSSION TITLE
Grafana: Invalid user or password

### DISCUSSION THREAD

**Discussion by MortenDecker** [tillsteinbach/VWsFriend#92] [Q&A] [answered]
I'm sure I have missed something very obvious but I simply cannot figure out what email/username to use in the http://IP-ADDRESS:3000 logon screen? I can connect successfully to WeConnect and pull car data but when trying to access to Visualization page I'm stuck at the login window. 
I've tried setting the USERNAME, PASSWORD parameters in the env file, using my grafana enabled email/password (not that I expected it to work but still) and now running out of ideas.
Thanks :)
Morten

**Comment by tillsteinbach** [comment]
setting USERNAME and PASSWORD is right. I don't know if it will be overwritten in grafana if you change it later. So you could try if admin/secret is working as password/user.
If it is, you can probably set your desired user and password in the env file and then delete the grafana volume and create it again with:
`docker-compose --env-file ./myconfig.env down -v grafana`

**Reply by MortenDecker** [reply]
Thanks for the fast turn-around.
I tried to start over. delete all volumes and set the USERNAME and PASSWORD in the env file and then start the image again (docker-compose) and this is what I get (like before)

![image](https://user-images.githubusercontent.com/4226931/139095498-3dbb7f65-04b1-4dd0-a214-dede68c43245.png)

**Reply by tillsteinbach** [reply]
Really strange. Maybe any fancy symbols in user or password?

**Reply by MortenDecker** [reply]
Just simple characters (letters and numbers). Well, I'll play around with it and let you know if I find the reason. Thanks for the help and clarifying where to set the username and password ;)

**Reply by tillsteinbach** [reply]
does login with admin/admin work?

**Reply by kennls** [reply]
I have exactly the same problem. Installed it 3 times, with the same problem.

It doesn't work with admin/admin or admin/secret.

**Reply by tillsteinbach** [reply]
Hmm, looks like this is really an issue. Unfortunately I cannot reproduce it here yet.

**Reply by MattiJarvinen** [reply]
Could it be docker version related?

Another might just be the usernames and the passwords used. At one project there was a configuration error if password contained shell specific characters $ @ etc. The password was clipped at that character. I do not recall what was the actual error in the docker-compose file or did that require setting the variable in a different way in the .env-file.

@kennls  & @MortenDecker 

By running `docker-compose config` current variables can be checked without running the containers. Look for GF_SECURITY_ADMIN_PASSWORD & GF_SECURITY_ADMIN_USER in the output. Those are your grafana password and username.

**Reply by tillsteinbach** [reply]
You can also find out the name of your grafana container with `docker ps` and then look how the variable resolves with:
`docker exec vwsfriend_grafana_1 bash -c 'echo "$GF_SECURITY_ADMIN_USER"'`
`docker exec vwsfriend_grafana_1 bash -c 'echo "$GF_SECURITY_ADMIN_PASSWORD"'`
you have to replace "vwsfriend_grafana_1" with the name you found with `docker ps`

**Comment by kennls** [comment]
> I'm sure I have missed something very obvious but I simply cannot figure out what email/username to use in the http://IP-ADDRESS:3000 logon screen? I can connect successfully to WeConnect and pull car data but when trying to access to Visualization page I'm stuck at the login window. 
> 
> I've tried setting the USERNAME, PASSWORD parameters in the env file, using my grafana enabled email/password (not that I expected it to work but still) and now running out of ideas.
> 
> Thanks :)
> 
> Morten
> 
> 

I found on mine, that the username is computername and not admin :)

**Comment by tillsteinbach** [comment, marked as answer]
This one should be fixed with the latest docker-compose.yml and .env template files
